File: 1302-deepest-leaves-sum/1302-deepest-leaves-sum.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int deepestLeavesSum(TreeNode root) {
        dfs(root,0,0);
        
         // Set set = map.entrySet();
         //  // Get an iterator
         //  Iterator i = set.iterator();
         //  // Display elements
         //  while(i.hasNext()) {
         //     Map.Entry me = (Map.Entry)i.next();
         //     System.out.print(me.getKey() + ": ");
         //     System.out.println(me.getValue());
         //  }
      
        return map.get(maxdepth);
    }
    HashMap<Integer, Integer> map = new HashMap<>();
    
    int maxdepth = Integer.MIN_VALUE;
    
    
    public void dfs(TreeNode root, int depth, int position){
        if(root == null)return;
        if(!map.containsKey(depth)){
            map.put(depth,root.val);
            maxdepth=depth;
            //System.out.println(root.val);
            //sum+=root.val;
        }
        else{
            int sum=map.get(depth);
            sum=sum+root.val;
            map.put(depth,sum);
        }
       
        //maxWidth = Math.max(maxWidth, position-map.get(depth)+1);
        dfs(root.left, depth+1, 2*position);
        dfs(root.right,depth+1, 2*position+1);
        
    }
}

File: 1302-deepest-leaves-sum/1302-deepest-leaves-sum.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
   queue<TreeNode*> q; 
public:
    int deepestLeavesSum(TreeNode* root) {
        int sum=0;
        q.push(root);
        while(!q.empty()){
            int size=q.size();
            sum=0;
            for(int i=0;i<size;i++){
                TreeNode* p=q.front();
                q.pop();
                
                sum+=p->val;
                if(p->right){
                   q.push(p->right); 
                }
                if(p->left){
                   q.push(p->left); 
                }
            }
        }
        return sum;
    }
};